server/enterprise/elasticsearch/elasticsearch/sortcombinations.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.enterprise for license information.
+
+package elasticsearch
+
+// This is a copy of https://github.com/elastic/go-elasticsearch/blob/9.0/typedapi/esdsl/sortcombinations.go.
+// This API is meant to be part of 9.0, but it got added to as part of 8.18.
```

Comment:
Wow, this is not good

Reply:
@agnivade Thoughts on keeping the version locked until there's a proper fix? Are we in a rush to update this dependency?

Reply:
It's a hack either ways. Either we add this, or we add another replace block to skip it in future. Usually, I'd prefer to keep dependencies up to date.

In any case, the work is already done though. So 0/5 from my side.

Reply:
Fair enough. I think the argument to keep it as is is that it feels a slightly safer option. If they are making breaking/substantial changes in minor versions, do we have confidence in the rest of the functionality to be untouched?

Reply:
Yeah I think the the tests are solid. The API change indeed broke some tests so there's some amount of confidence at least :)